core/rawdb/accessors_chain.go

```diff
@@ -1019,20 +1019,23 @@ func WriteBlock(db kv.RwTx, block *types.Block) error {
 // keeps genesis in db: [1, to)
 // doesn't change sequences of kv.EthTx and kv.NonCanonicalTxs
 // doesn't delete Receipts, Senders, Canonical markers, TotalDifficulty
-func PruneBlocks(tx kv.RwTx, blockTo uint64, blocksDeleteLimit int) error {
+// Returns true if there is nothing to prune
+func PruneBlocks(tx kv.RwTx, blockTo uint64, blocksDeleteLimit int) (bool, error) {
+	nothingToPrune := false
```

Comment:
golang's style api is `val, ok := someMap[key]` - it returns "exists" instead of "notExists" bool.
also if return some non-trivial bool better add name of variable to func signature `(nothingToPrune bool, err error) {` then future users of func can understand what it returns without diving-in.

Reply:
changed to existBlocksToPrune variable which is true by default (I suppose that we assume that there is something to prune until we prune enough)